GitHubTeams.ps1

```diff
@@ -276,6 +313,484 @@ filter Get-GitHubTeamMember
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .OUTPUTS
+        None
+
+    .EXAMPLE
+        Remove-GitHubTeam -OrganizationName PowerShell -TeamName Developers
+
+        Removes the 'Developers' GitHub team from the 'PowerShell' organization.
+
+    .EXAMPLE
+        Remove-GitHubTeam -OrganizationName PowerShell -TeamName Developers -Force
+
+        Removes the 'Developers' GitHub team from the 'PowerShell' organization without prompting.
+
+    .EXAMPLE
+        $team = Get-GitHubTeam -OrganizationName PowerShell -TeamName Developers
+        $team | Remove-GitHubTeam -Force
+
+        You can also pipe in a GitHub team that was returned from a previous command.
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSShouldProcess', '',
+        Justification = 'Methods called within here make use of PSShouldProcess, and the switch
+        is passed on to them inherently.')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false,
+        ConfirmImpact = 'High'
+    )]
+    [Alias('Delete-GitHubTeam')]
+    param
+    (
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName,
+            Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline,
+            ValueFromPipelineByPropertyName,
+            Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [switch] $Force,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $RepositoryName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+    }
+
+    $getGitHubTeamParms = @{
+        OrganizationName = $OrganizationName
+        TeamName = $TeamName
+        Whatif = $false
+        Confirm = $false
+    }
+    if ($PSBoundParameters.ContainsKey('AccessToken'))
+    {
+        $getGitHubTeamParms['AccessToken'] = $AccessToken
+    }
+    if ($PSBoundParameters.ContainsKey('NoStatus'))
+    {
+        $getGitHubTeamParms['NoStatus'] = $NoStatus
+    }
+
+    $team = Get-GitHubTeam @getGitHubTeamParms
+
+    $uriFragment = "/orgs/$OrganizationName/teams/$($team.slug)"
```

Comment:
Similar feedback for New/Set.  We should enable the ability to call this function without the need for the additional Get to retrieve the slug.  If a user pipes in an existing team to this function, it should have all the information it needs to be able to delete it.  That means that the Slug needs to be an alternative parameterset for this function.

Reply:
I really don't see why. They user of the function has no knowledge of what a `TeamSlug` is. What is the problem with the function resolving the name for them?

Reply:
I ended up [explaining this](https://github.com/microsoft/PowerShellForGitHub/pull/238#discussion_r443268865) in your #238 PR.  Repeating it here though:

The Microsoft org has **2100 teams** (and that number continues to grow).

* Getting all of them with `Get-GitHubTeam -OrganizationName microsoft -NoStatus` adds an additional **22 seconds** to method execution.
* Calling it _with_ status (e.g.  `Get-GitHubTeam -OrganizationName microsoft`)  adds an additional **104** seconds.
* _Allowing_ users to call into the method providing the slug for the team would add an additional **0 seconds**. 

I agree that exposing the existence of slugs makes the signature a bit uglier, however _not_ exposing the option of using slugs means that the API call can needlessly take additional _minutes_ to complete.

Reply:
So what this tells me is first, disable by default that horrendous code which is Status, and second, look at the inefficiencies of `Invoke-GHRestMethod` and `Invoke-GHRestMethodMultipleResult`, performing multiple update checks, telemetry calls etc and maybe you will get that time down to less than 5 seconds.

Reply:
Let's just put the status discussion to the side for the moment and focus on the `-NoStatus` time.

* `Invoke-UpdateCheck` takes on average **2 ms**.  (Remember, all it's doing after the first check for the day is a call to `Get-Date` and a single `if` comparison.  The added load to normal module functionality is minimal at best as the average run time demonstrates.

* Disabling telemetry shaved off **1 second** on average from the total.  That's not the silver bullet.  That being said, there is food for thought there to determine if enough value is being gained from having a telemetry event for each `Invoke-GHRestMethod` within the confines of a `Invoke-GHRestMethodMultipleResult`.  I'll need to consider that one a bit more separately.

They key thing to keep in mind here is that to retrieve 2100 teams it has to make **70** REST requests.

Let's look at this totally bare bones:

```powershell
Measure-Command {
      $ProgressPreference = 'SilentlyContinue'
      1..70 |% { 
          $null = Invoke-WebRequest -Headers $headers -UseDefaultCredentials -UseBasicParsing -Method Get -Uri 'https://api.github.com/organizations/$orgId/teams?page=$_'
      }
}
```

So, using pure basic PowerShell with nothing from this module takes **17.5 seconds** on average.  Which is **4.5 seconds** faster than what the module is doing.  Those extra few seconds are buying us:
  * Telemetry
  * Smart Objects (so, actual DateTime objects instead of date strings)
  * Additional properties and type names being added
  * Logging
  * Maybe some other stuff that I'm missing.

I'm not seeing any way that this call could get down to 5 seconds, let alone under 17.5 seconds.

I'm also not seeing real inefficiencies that `Invoke-GHRestMethod` is causing in the normal case.  If you are, then please point them out so that they can get fixed.

I do continue to see the opportunity for this call to be avoided entirely however in the scenario where someone chooses to pipe in the team object instead.  If they choose to pass in just a TeamName, then they will have to live with a longer execution time due to the reality of the need to get the slug (it would be no different than having done the same work that you're doing, but external to this function and then piping in the result of a `Where-Object`)...but if the user has the slug, I still see no reason to not give them the opportunity for faster execution and avoid the `Get`.

Reply:
I did a little more testing for completeness:

```powershell
Set-GitHubConfiguration -SessionOnly -DefaultNoStatus -DisableLogging -DisableSmarterObjects -DisablePipelineSupport

Measure-Command { $null = Get-GitHubTeam -OrganizationName microsoft }
```

Came in, on average around **18.5 seconds**.  Adding in `-DisableTelemetry` reduced it by another **.5 second** or so.

So, `Invoke-GHRestMethod` is able to match the performance of raw `Invoke-WebRequest` in this scenario within around .5 second overall with no code change.  There might be some limited opportunities to make it slightly faster, but I don't believe that there's much in the way of negative module stuff getting in the way here.  Each of those things that were disabled in that test have demonstrable value, _and_ each of them has a clearly documented way to be disabled in the event that a user values slight performance gains in lieu of the added benefits those disabled features would have provided.